src/librustc_typeck/diagnostics.rs

```diff
@@ -170,6 +170,38 @@ Since `return;` is just like `return ();`, there is a mismatch between the
 function's return type and the value being returned.
 "##,
 
+E0070: r##"
+The left-hand side of an assignment operator must be an lvalue expression. An
+lvalue expression represents a memory location and includes item paths (ie,
+namespaced variables), dereferences, indexing expressions, and field references.
+
+More details can be found here:
+https://doc.rust-lang.org/reference.html#lvalues,-rvalues-and-temporaries
+
+Now, we can go further. Here are some bad examples:
+```
+const SOME_CONST : i32 = 12;
+
+fn some_function() {
+    SOME_CONST = 14; // error !
+    1=3; // error 
+}
+```
+
+And now let's give good examples:
```

Comment:
Well, I don't know if pointing people at `static mut` is necessarily what we want to do here, as opposed to a non-mut `static` that holds an appropriate atomic data type... we *can* point people at `static mut`, but then they'll have to use unsafe code to access it (for good reason) ...